### ISSUE TITLE
Weird methods in ElasticsearchClient

### ISSUE THREAD

**Issue by darxriggs** [elastic/elasticsearch-java#145]
There are methods in `ElasticsearchClient` that look like not very useful.

They do not take any parameter, therefore using the builder internally without setting any value. When called, they raise an `IllegalStateException`.

Examples
- [public BulkResponse bulk()](https://github.com/elastic/elasticsearch-java/blob/e6d2abe7b98135b5885c218ef0c7e55cb8f93ac1/java-client/src/main/java/co/elastic/clients/elasticsearch/ElasticsearchClient.java#L339-L342)
- [public MtermvectorsResponse mtermvectors()](https://github.com/elastic/elasticsearch-java/blob/e6d2abe7b98135b5885c218ef0c7e55cb8f93ac1/java-client/src/main/java/co/elastic/clients/elasticsearch/ElasticsearchClient.java#L1138-L1141)
- [public FieldCapsResponse fieldCaps()](https://github.com/elastic/elasticsearch-java/blob/e6d2abe7b98135b5885c218ef0c7e55cb8f93ac1/java-client/src/main/java/co/elastic/clients/elasticsearch/ElasticsearchClient.java#L800-L803)
- [public ReindexResponse reindex()](https://github.com/elastic/elasticsearch-java/blob/e6d2abe7b98135b5885c218ef0c7e55cb8f93ac1/java-client/src/main/java/co/elastic/clients/elasticsearch/ElasticsearchClient.java#L1306-L1309)
- [public RenderSearchTemplateResponse renderSearchTemplate()](https://github.com/elastic/elasticsearch-java/blob/e6d2abe7b98135b5885c218ef0c7e55cb8f93ac1/java-client/src/main/java/co/elastic/clients/elasticsearch/ElasticsearchClient.java#L1389-L1392)

Looking at the API documentation, there is a least some mandatory parameter.

Is this maybe the result of generating the client based on the API specification and they should better be removed?

**Comment by swallez** [comment]
@darxriggs you're right. The code generator produces a no-parameter version of API methods when all properties of the request object are optional. For some of requests that was an error as some properties are actually required (see specification PRs linked above) but for some others at least one of these properties must be set.

We'll take a different approach and generate these no-parameter methods from an opt-in list where we know that this is indeed correct. In the meantime, well, avoid using these 😉 

And thank you for having reported this!
